compile/events/eventManagerInterface.cpp

```diff
@@ -136,16 +137,14 @@ void mexFunction(int nlhs, mxArray *plhs[], int nrhs, const mxArray *prhs[])
 
     // register
     if (!strcmp("register", cmd)) {
-        for ( int type = MESSAGE; type <= PLOT; ++type){
-            if (type == eventType){
-   	            void (*cbPtr)(const baseEvent&) = eventCallback;
-                initDyLib();
-                auto addListener = library->get_function<void(enum eventTypes, void (*)(const baseEvent&))>("addListener");
-                addListener((enum eventTypes)eventType, cbPtr);
-                return;
-            }
-        }
-        mexErrMsgTxt("Event type not recognized.");
+       if (eventType < 0 || eventType > static_cast<int>(EventTypes::Plot)) {
```

Comment:
Is there not a better test here? It's very easy to add to/modify this enum down the line and plot may not be the final entry. Is there a way of obtaining the number of options?

Reply:
To the best of my knowledge, there is no way to get the count. One solution I've seen is to add an extra enum e.g. END which marks the end and will not be moved

Reply:
Hmmm, not spectacular either way then.

Reply:
I will revisit if I find a better solution